lib/AfricasTalking/Airtime.rb

```diff
@@ -1,75 +1,75 @@
+      item['amount'].to_s.prepend(item['currencyCode'].to_s + ' ')
+      item.delete('currencyCode')
+    end
+    post_body = {
+      'username' => @username,
+      'recipients' => recipients.to_json
+    }
+    response = sendNormalRequest(url, post_body)
+    if @response_code == HTTP_CREATED
+      responses = JSON.parse(response, quirky_mode: true)
+      if !responses['responses'].empty?
+        results = responses['responses'].collect do |result|
+          AirtimeResponse.new result['status'], result['phoneNumber'], result['amount'], result['requestId'], result['errorMessage'], result['discount']
+        end
+        return SendAirtimeResult.new responses['errorMessage'], responses['numSent'], responses['totalAmount'], responses['totalDiscount'], results
+      else
+        raise AfricasTalkingException, responses['errorMessage']
+      end
+      raise AfricasTalkingException, response
+    end
+  end
 
-		recipients = options.each{|item| 
-			validateParamsPresence? item, %w(phoneNumber currencyCode amount)
-			item['amount'].to_s.prepend(item['currencyCode'].to_s + " ") 
-			item.delete('currencyCode')
-		}
-		post_body = {
-					'username'   => @username,
-					'recipients' => recipients.to_json,
-				}
-		response = sendNormalRequest(url, post_body) 
-		if (@response_code == HTTP_CREATED)
-			responses = JSON.parse(response, :quirky_mode =>true)
-			if (responses['responses'].length > 0)
-				# 
-				results = responses['responses'].collect{ |result|
-					# 
-					AirtimeResponse.new result['status'], result['phoneNumber'],result['amount'],result['requestId'], result['errorMessage'], result['discount']
-				}
-				# 
-				return SendAirtimeResult.new responses["errorMessage"], responses["numSent"], responses["totalAmount"], responses["totalDiscount"], results
-			else
-				raise AfricasTalkingException, responses['errorMessage']
-			end
-			raise AfricasTalkingException, response
-		end
-	end
+  private
 
-	private
-		def getAirtimeUrl()
-			return getApiHost() + "/version1/airtime"
-		end
+  def getAirtimeUrl
+    getApiHost + '/version1/airtime'
+  end
 
-		def getApiHost()
-			if(@username == "sandbox")
-				return "https://api.sandbox.africastalking.com"
-			else
-				return "https://api.africastalking.com"
-			end
-		end
+  def getApiHost
+    if @username == 'sandbox'
+      'https://api.sandbox.africastalking.com'
+    else
+      'https://api.africastalking.com'
+    end
+  end
 end
 class AirtimeResponse
-	attr_reader :amount, :phoneNumber, :requestId, :status, :errorMessage, :discount
-	def initialize(status_, number_, amount_, requestId_, errorMessage_, discount_)
-		@status       = status_
-		@phoneNumber  = number_
-		@amount       = amount_
-		@requestId    = requestId_
-		@errorMessage = errorMessage_
-		@discount     = discount_
-	end
+  attr_reader :amount, :phoneNumber, :requestId, :status, :errorMessage, :discount
+  def initialize(status_, number_, amount_, requestId_, errorMessage_, discount_)
```

Comment:
That's a subjective point, but wouldn't it be better to use something like this, with less boilerplate code? :slightly_smiling_face: 
```rb
AirtimeResponse = Struct.new(:amount, :phoneNumber, :requestId, :status, :errorMessage, :discount)
```

Reply:
This is more flexible since it provides three way of accessing and modifying member values.
`accessor_method`, `string` and `symbol`